src/include/function/cast/vector_cast_operations.h

```diff
@@ -20,6 +20,9 @@ class VectorCastOperations : public VectorOperations {
         assert(params.size() == 1);
         UnaryOperationExecutor::executeCast<OPERAND_TYPE, RESULT_TYPE, FUNC>(*params[0], result);
     }
+
+    static scalar_exec_func getCastExecFunc(
```

Comment:
Since you already implement CastToDouble operation. I think we should also expose to_double() function for the user. So instead of writing this `getCastExecFunc`. A better solution is to follow what `CastToDateVectorOperation` did and register into catalog so that we can get to_double() as well.

Reply:
changed as we discussed.